adapta/storage/distributed_object_store/v3/datastax_astra/astra_client.py

```diff
@@ -304,21 +300,18 @@ def to_frame(
             self._session is not None
         ), "Please instantiate an AstraClient using with AstraClient(...) before calling this method"
 
-        select_columns = (
-            list(map(normalize_column_name, select_columns))
-            if select_columns
-            else [f.name for f in fields(model_class)]
-        )
+        select_columns = list(map(normalize_column_name, select_columns)) if select_columns else None
 
-        model_class: Type[Model] = self._model_dataclass(
-            value=model_class,
+        # pylint: disable=too-many-function-args
+        model_class = model_mapper_factory(
```

Comment:
Let's maybe name this cassandra_model? So we do not touch the input var, also better readability